resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/LocalDirsFeatureStep.scala

```diff
@@ -43,20 +44,34 @@ private[spark] class LocalDirsFeatureStep(
     val localDirVolumes = resolvedLocalDirs
       .zipWithIndex
       .map { case (localDir, index) =>
-        new VolumeBuilder()
-          .withName(s"spark-local-dir-${index + 1}")
-          .withNewEmptyDir()
-            .withMedium(if (useLocalDirTmpFs) "Memory" else null)
-          .endEmptyDir()
-          .build()
+        val name = s"spark-local-dir-${index + 1}"
+        hasVolume(pod, name) match {
```

Comment:
not a super big deal, but `hasVolume` is basically the same logic `pod.pod.getSpec().getVolumes().asScala.exists` as ` pod.pod.getSpec.getVolumes().asScala.find`

I think you can refactor this to avoid scanning multiple times.

Reply:
Updated in the latest commit.